mne/dipole.py

```diff
@@ -354,7 +356,19 @@ def constraint(rd):
                                         return_dists=True)[1][0]
                 return -dist
             else:
-                return 1.
+
+                # Once we know the dipole is below the inner skull,
+                # let's check if its distance to the inner skull is at least
+                # min_dist_to_inner_skull. If it is not, it is penalized
+                # proportional to its distance.
+
+                dist = _compute_nearest(surf['rr'], rd[np.newaxis, :],
+                                        return_dists=True)[1][0]
```

Comment:
Maybe computation should only be done if `dist is not None`, and `dist` should default to `None` for consistency with old behavior... I'm not sure. Depends on if this is considered a bug or not.